packages/gateway/src/index.js

```diff
@@ -0,0 +1,43 @@
+import { addCorsHeaders } from './cors.js'
+import { errorHandler } from './error-handler.js'
+import { getCidFromSubdomainUrl } from './utils/cid.js'
+
+/**
+ * @typedef {Object} Env
+ * @property {string} IPFS_GATEWAY
+ */
+
+/**
+ * Handle gateway request
+ * @param {Request} request
+ * @param {Env} env
+ */
+async function handleRequest(request, env) {
+  const publicGatewayUrl = new URL('ipfs', env.IPFS_GATEWAY)
+  const url = new URL(request.url)
+  const cid = getCidFromSubdomainUrl(url.hostname)
+  const response = await fetch(
+    `${publicGatewayUrl.toString()}/${cid}${url.pathname || ''}`
+  )
```

Comment:
I _think_ just putting it in a string will automatically call `toString`...

```suggestion
  const response = await fetch(`${publicGatewayUrl}/${cid}${url.pathname || ''}`)
```